### PR TITLE
Fixing filter on product without price.

### PR DESCRIPTION
**Pull request by christopherhero**

### CODE REVIEW

**Review by leszczuu — APPROVED** [review]

File: spec/QueryBuilder/HasPriceBetweenQueryBuilderSpec.php [hunk 1]
```diff
@@ -106,4 +106,85 @@ function it_converts_fractional_currency_properly(
...
+        $currency->getCode()->willReturn('USD');
+        $currencyContext->getCurrencyCode()->willReturn('USD');
+
+        $channelPricingNameResolver->resolvePropertyName('web')->willReturn('web');
+
+        $range = $this->buildQuery([]);
+
+        $range->shouldBeNull();
+    }
 }
```
Comment (tgrochowski) [inline spec/QueryBuilder/HasPriceBetweenQueryBuilderSpec.php:190]: Missing new line

File: src/QueryBuilder/HasPriceBetweenQueryBuilder.php [hunk 2]
```diff
@@ -94,4 +97,20 @@ private function convertFromString(string $price): int
...
+    {
+        foreach (['gte' => $min, 'lte' => $max] as $key => $value) {
+            if (null !== $value) {
+                $paramValue[$key] = $value;
+            }
+        }
+
+        return $paramValue ?? null;
+    }
 }
```
Comment (tgrochowski) [inline src/QueryBuilder/HasPriceBetweenQueryBuilder.php:116]: Missing new line

File: spec/QueryBuilder/HasPriceBetweenQueryBuilderSpec.php (hunk 1 above)
Comment (christopherhero) [reply spec/QueryBuilder/HasPriceBetweenQueryBuilderSpec.php:190]: it is but on GH is not visible :)
(you can check when you switch on branch)

File: src/QueryBuilder/HasPriceBetweenQueryBuilder.php (hunk 2 above)
Comment (christopherhero) [reply src/QueryBuilder/HasPriceBetweenQueryBuilder.php:116]: it is but on GH is not visible :)
(you can check when you switch on branch)

**Review by tgrochowski — APPROVED** [review]

**Pull request merged**